### ISSUE TITLE
autoscaler for cluster xxxxxx might already  exists.

### ISSUE THREAD

**Issue by bobby0724** [terraform-redhat/terraform-rhcs-rosa-hcp#12]
Hello i am haviing this issue after destroyin my cluster and trying to recreate

module.rosa_cluster_hcp.rhcs_cluster_rosa_hcp.rosa_hcp_cluster: Modifying... [id=2as8ja8fdlblueccm19sehb6s6d0m40t]
module.rosa_cluster_hcp.rhcs_cluster_rosa_hcp.rosa_hcp_cluster: Modifications complete after 1s [id=2as8ja8fdlblueccm19sehb6s6d0m40t]
module.rosa_cluster_hcp.rhcs_hcp_cluster_autoscaler.cluster_autoscaler[0]: Creating...
Error: Can't create autoscaler
with module.rosa_cluster_hcp.rhcs_hcp_cluster_autoscaler.cluster_autoscaler[0],
on .terraform/modules/rosa_cluster_hcp/modules/rosa-cluster-hcp/main.tf line 113, in resource "rhcs_hcp_cluster_autoscaler" "cluster_autoscaler":
113: resource "rhcs_hcp_cluster_autoscaler" "cluster_autoscaler" {
Autoscaler for cluster '2as8ja8fdlblueccm19sehb6s6d0m40t' might already
exists. Error: status is 403, identifier is '403', code is
'CLUSTERS-MGMT-403' and operation identifier is
'76f0df43-0be9-434d-8f1e-3cd1f4ae4c01': Autoscaler configuration is not
available

**Comment by gdbranco** [comment]
The resource on TF is already prepared. However, the autoscaler functionality is not yet available publicly for HCP

**Comment by davivcgarcia** [comment]
Is this `cluster_autoscaler` feature the same that we see in the OCM UI when creating ROSA HCP?

![Screenshot 2024-10-04 at 17 44 11](https://github.com/user-attachments/assets/8edd4ec7-49f5-4ec2-a3dd-3c2d13b86635)

**Comment by gdbranco** [comment]
@davivcgarcia No, this is a cluster autoscaler. That one currently in the UI is the autoscaling parameters for the initial instance pool for customer workload that you get when installing the cluster. The cluster autoscaler has many more options in comparison to min and max nodes

**Comment by davivcgarcia** [comment]
@gdbranco Do you have an example of how to setup the default worker machine pool autoscaling parameters at cluster creation? And what is the difference between the cluster autoscaler and the autoscaling parameters from the machine pools?

**Comment by amandahla** [comment]
You can find an example of rhcs_cluster_autoscaler in:
https://registry.terraform.io/providers/terraform-redhat/rhcs/latest/docs/resources/cluster_autoscaler
And autoscaling:
https://registry.terraform.io/providers/terraform-redhat/rhcs/latest/docs/resources/hcp_machine_pool#nestedatt--autoscaling

In short, the cluster autoscaler is the mechanism that decides when and why to scale based on global pod and resource metrics. 
The machine pool autoscaling parameters define where the scaling is allowed to happen and the strict minimum and maximum boundaries the cluster autoscaler must respect for that specific group of machines.
